app/models/miq_server/environment_management.rb

```diff
@@ -43,9 +43,9 @@ def get_network_information
           hostname    = LinuxAdmin::Hosts.new.hostname
           mac_address = eth0.mac_address
         else
-          require 'MiqSockUtil'
-          ipaddr      = MiqSockUtil.getIpAddr
-          hostname    = MiqSockUtil.getFullyQualifiedDomainName
+          require 'socket'
+          ipaddr      = Socket.ip_address_list.select{ |addr| addr.ipv4? && !addr.ipv4_private? }.first&.ip_address
+          hostname    = Socket.gethostbyname(Socket.gethostname).first
```

Comment:
Interesting.  Just looking at this, I wondered if maybe this is the same code that drb uses to resolve the drb server URI that caused me countless time figuring out why the hostname resolution failed so miserably in another issue:

https://github.com/ManageIQ/manageiq-automation_engine/pull/431

and yeah, it looks like drb uses `Socket.gethostname`: https://github.com/ruby/ruby/blob/v2_6_5/lib/drb/drb.rb#L879-L884

Do we use this elsewhere?  We hit that drb issues at customers before so I wonder if this something to be wary of.  It's possible `Socket::getaddrinfo` is the real problem as we didn't determine where the problem was DRb's `getservername`method:  https://github.com/ruby/ruby/blob/v2_6_5/lib/drb/drb.rb#L847-L858

Reply:
@jrafanie I think `getaddrinfo` is the preferred approach these days, but this code is only called when it's NOT an appliance, so I'm not sure how critical it is here. I just copied the one-line from MiqSockUtil in this case. But, if you have a better alternative, I'm listening.

Reply:
Does _not_ appliance imply containers or not? (in addition to local dev)

Reply:
@Fryguy Good question. I'm just imitating the current behavior, and actually I was thinking maybe this should add a check for `Rails.development?`, because otherwise I think you *would* want the private IP's (only).

On my laptop the old code `MiqSockUtil.getIpAddr` returns `127.0.0.1` and not one of my private IPs.

Reply:
I just checked and `echo $APPLIANCE` returns true inside of a container, so this is completely local code.

Reply:
Ok, then I think what we have is good.

Reply:
Ugh, we should make this conditional be `is_production_build?` if it's supposed to be for appliance, podified or containers.

Reply:
Agreed, but that should not be in this PR